content/geoip2/web-services/using-the-javascript-client.mdx

```diff
@@ -0,0 +1,241 @@
+Explorer 8, this is implemented as a non-enumerable property using
+`defineProperty`.
+
+* If there is an error, the function calls the `onError` with the error object
+
+as a parameter.
+
+* The optional `options` parameter is an object containing flags for the
+
+service. This parameter is reserved for future use. There are no options at
+this time.
+
+### Example: Display the user's city name in a page
+
+In this example, we are displaying the user's city name in a page.  We have 2
+files: page.html and demo.js.
+
+```html
+<!-- page.html -->
+<!doctype html>
+<html>
+  <head>
+    <title>Which city am I in?</title>
+    <script src="//geoip-js.com/js/apis/geoip2/v2.1/geoip2.js"></script>
+    <!-- make sure this is after the geoip2.js file -->
+    <script src="/demo.js"></script>
+  </head>
+
+  <body>
+    <p>
+       Where am I?
+    </p>
+
+    <p>
+      You are in <span id="city"></span>.
+    </p>
+  </body>
+</html>
+```
+
+```js
+// demo.js
+var fillInPage = (function() {
+    var updateCityText = function(geoipResponse) {
+
+        /* It's possible that we won't have any names for this city.
+           For language codes with a special character such as pt-BR,
+           replace names.en with names['pt-BR'].
+        */
+        var cityName = geoipResponse.city.names.en || 'your city';
+
+        document.getElementById('city').innerHTML = cityName
+    };
+
+    var onSuccess = function(geoipResponse) {
+        updateCityText(geoipResponse);
+    };
+
+    /* If we get an error we will display an error message*/
+    var onError = function(error) {
+        document.getElementById('city').innerHTML = 'an error!  Please try again..'
+    };
+
+    return function() {
+        if (typeof geoip2 !== 'undefined') {
+            geoip2.city(onSuccess, onError);
+        } else {
+            document.getElementById('city').innerHTML = 'a browser that blocks GeoIP2 requests'
+        }
+    };
+}());
+
+fillInPage();
+```
+
+### Successful responses
+
+A successful response consists of a single JavaScript object. That object
+contains a set of key/value pairs, where the keys are things like `country` and
+`traits`, and the values are objects or arrays of objects.
+
+While our REST API is documented as omitting keys, for the JavaScript API, we
+add missing arrays and objects. This makes it much simpler to work with the
+response, as you will not need to check whether the data structure exists
+before using it.
+
+Wherever a property’s value could be an object or array, we will add an empty
+object or array as needed. For example, if you call the `city()` method and the
+response from the MaxMind server has no `city` key at all, it will end up being
+filled in as:
+
+```json
+{
+  "city":     {
+      "names": {}
+  },
+  "country":   ...,
+  "continent": ...,
+  ...
+}
+```
+
+If the response contains no `subdivisions` key, you will get this:
+
+```json
+{
+  "city":         ...,
```

Comment:
Not a huge deal, perhaps more of a personal preference. Generally, I'm not a fan of aligning properties like this. It makes maintaining a little harder when the widest property changes.